libraries/AP_HAL_PX4/CAN.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2015-2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "AP_HAL_PX4.h"
+#include <systemlib/perf_counter.h>
+#include <AP_HAL/CAN.h>
+
+class PX4::PX4CAN : public AP_HAL::CAN {
+public:
+    PX4CAN() { }
+    ~PX4CAN() { }
+    int16_t send(const uavcan::CanFrame& frame, uavcan::MonotonicTime tx_deadline, uavcan::CanIOFlags flags);
+    int16_t receive(uavcan::CanFrame& out_frame, uavcan::MonotonicTime& out_ts_monotonic, uavcan::UtcTime& out_ts_utc,
+            uavcan::CanIOFlags& out_flags);
+    int16_t configureFilters(const uavcan::CanFilterConfig* filter_configs, uint16_t num_configs);
+    uint16_t getNumFilters() const;
+    uint64_t getErrorCount() const;
+    bool begin(uint32_t bitrate);
+    bool begin(uint32_t bitrate, float samplepoint);
+    void end();
+    void reset();
+    bool is_initialized();
+    int32_t tx_pending();
+    uint32_t available();
+
+private:
+    uint32_t _baudrate;
+    volatile bool _initialised;
+    volatile bool _in_timer;
+    perf_counter_t  _perf_can;
```

Comment:
oh... again, this "problem". It doesn't **need** to be initialized to 0 because (a) we override the new() operator globally to zero memory it allocates before calling the constructor and (b) code that is static is going to be zeroed.  This doesn't apply if the object is going to be instantiated on stack, which doesn't seem to be the case.

Why this is done:  to avoid wasting some precious KB of flash on all initialization to 0 (C++11 or initialization lists, it doesn't matter). I actually don't think we should keep doing this because tools for static analysis don't understand this, and it makes a real pain to use them with these false positives.

Long story short: 1) it's ok to leave as is;  2) for me it'd be better to initialize the fields and 3) this is subject to different opinions among the devs.